### PR TITLE
Change products template to use patterns

### PR DESCRIPTION
**Pull request by livnat-a**
Change products template to use patterns library

### CODE REVIEW

**Review by matveyok — APPROVED** [review]
lgtm - didn't run it locally tho so I hope everything works :)

File: custom-products-catalog/template/src/dashboard/pages/page.tsx [hunk 1]
```diff
@@ -1,161 +1,195 @@
...
+                  submit: (products: products.Product[]) => (
+                    Promise.all(
+                      products.map((product: products.Product) => deleteProduct(product._id))
+                    )
+                  ),
+                  successToast: {
+                    message: `${_product.name} deleted successfully`,
+                    type: 'SUCCESS',
+                  },
+                  errorToast: () => 'An error',
```
Comment (omrimn) [inline custom-products-catalog/template/src/dashboard/pages/page.tsx:185]: maybe this message should be more specific? 

`Product deletion failed` ?

File: custom-products-catalog/template/src/dashboard/hooks/stores.ts [hunk 2]
```diff
@@ -1,89 +1,57 @@
...
 
-export function useCreateProduct() {
-  const queryClient = useQueryClient();
-  const { showToast } = useDashboard();
-  const { createProduct } = useWixModules(products);
+  return useCallback((productName: string) => {
+    const newProduct = {
+      id: window.Date().toString(),
+      name: productName,
+      createdDate: new window.Date(),
```
Comment (omrimn) [inline custom-products-catalog/template/src/dashboard/hooks/stores.ts:13]: `window.` is not necessary.

File: custom-products-catalog/template/src/dashboard/pages/page.tsx [hunk 3]
```diff
@@ -1,161 +1,259 @@
...
-      setProductIdsToDelete(new Set(productIdsToDelete));
-    } else {
-      productIdsToDelete.add(productId);
-      setProductIdsToDelete(new Set(productIdsToDelete));
-    }
-  };
+  const optimisticActions = useOptimisticActions(tableState.collection, {
+    orderBy: () => [],
+    predicate: ({ search }) => {
+      return (product) => {
```
Comment (livnat-a) [inline custom-products-catalog/template/src/dashboard/pages/page.tsx:104]: @matveyok you need to also add here the filters logic

**Review by omrimn — APPROVED** [review]

**Pull request merged**